File: golang-for-nodejs-developers/examples/http_server.go
package main

import (
	"fmt"
	"net/http"
)

func handler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "text/plain")
	w.WriteHeader(http.StatusOK)
	w.Write([]byte("hello world"))
}

func main() {
	http.HandleFunc("/", handler)
	fmt.Println("HTTP server listening on port 8080")
	if err := http.ListenAndServe(":8080", nil); err != nil {
		panic(err)
	}
}

File: golang-for-nodejs-developers/examples/http_server.js
import http from 'node:http';

function handler(request, response) {
  response.writeHead(200, { 'Content-Type': 'text/plain' });
  response.write('hello world');
  response.end();
}

const server = http.createServer(handler);
server.listen(8080, () => {
  console.log('HTTP server listening on port 8080');
});
